becquerel/core/spectrum.py

```diff
@@ -973,12 +973,15 @@ def find_bin_index(self, x, use_kev=True):
 
         bin_edges = self.bin_edges_kev if use_kev else self.bin_edges_raw
         bin_widths = self.bin_widths_kev if use_kev else self.bin_widths_raw
+        x = np.array(x)
 
-        assert x >= bin_edges[0]
-        assert x < bin_edges[-1]
+        if np.any(x < bin_edges[0]):
+            raise ValueError('requested x is < lowest bin edge')
```

Comment:
use SpectrumError?

Reply:
There are a few other places in `spectrum.py` where a `ValueError` is raised rather than a `SpectrumError`, so I'm not sure what's the most consistent. I'll go with the latter for now.

Reply:
Yeah, I think those are all legacy...